src/autodiff.jl

```diff
@@ -382,6 +382,136 @@ function residual_jacobian!(arrays::Jacobian,
     return nothing
 end
 
+function residual_jacobian_adj!(arrays::Jacobian,
```

Comment:
I would maybe add a comment explaining that we are computing the full Jacobian using `m` calls to the adjoint code `residual_adj_polar!`

Reply:
And also that `dJ` will be evaluated as a dense matrix first.

Reply:
No `m` calls anymore. This code is gone too.